### PR TITLE
[tune] Change log handling for Tune

### LINKED ISSUE #3653: [tune] Result logs color
Hi all,

Thanks for creating Ray, Tune, and RLlib!

This may be a bit subjective, but I wanted to report that I am always a bit irritated by the red color of the Tune result logging output. Red always looks like a warning or error message to me. Does anyone else feel that way? 

Having said that, it's totally sensible to have the results emphasized in some color to make it nice and visible! Could it be green instead? I think that could make the Ray experience even more joyful!

Thanks & best regards,
Marlon

### PR DESCRIPTION
**Pull request by richardliaw**
Also provides a small retry mechanism for a transient error as reported
by #3340.

Closes #3653.

### CODE REVIEW

**Comment by richardliaw** [comment]
jenkins retest this please

File: python/ray/tune/__init__.py [hunk 1]
```diff
@@ -2,13 +2,30 @@
...
+
 from ray.tune.error import TuneError
 from ray.tune.tune import run_experiments
 from ray.tune.experiment import Experiment
 from ray.tune.registry import register_env, register_trainable
 from ray.tune.trainable import Trainable
 from ray.tune.suggest import grid_search, function, sample_from
 
+
+def _setup_logger():
```
Comment (ericl) [inline python/ray/tune/__init__.py]: Do we also need stream=sys.stdout for rllib?

File: python/ray/tune/trial.py [hunk 2]
```diff
@@ -302,8 +302,8 @@ def update_last_result(self, result, terminate=False):
             result.update(done=True)
         if self.verbose and (terminate or time.time() - self.last_debug >
                              DEBUG_PRINT_INTERVAL):
-            logger.info("Result for {}:".format(self))
-            logger.info("  {}".format(
+            logger.info("\nResult for {}:\n  {}".format(
+                self,
```
Comment (ericl) [inline python/ray/tune/trial.py]: I think some of these should become prints right?

File: python/ray/tune/__init__.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/tune/__init__.py]: Hm maybe I should just keep it at stderr and then move `logger.info` into `print` for the pretty result

**Comment by richardliaw** [comment]
ready for another review

**Review by ericl — APPROVED** [review]

**Pull request merged**